crates/solver/src/solver/single_order_solver.rs

```diff
@@ -310,7 +419,12 @@ impl SingleOrderSolver {
 
         let simulation = match result {
             Rating::Ok(simulation) => simulation,
-            Rating::Err(err) => return Err(err.error).context("failed to rate the settlement"),
+            Rating::Err(err) => {
+                return Err(err.error).context(format!(
+                    "failed to rate the settlement: {:?}",
+                    err.simulation.transaction
+                ))
+            }
```

Comment:
I don't think we should attach transaction information to the error message, as there is no guarantee that the caller will log it. We can also just:

```suggestion
            Rating::Err(err) => {
                tracing::warn!(?err, ?transaction, "failed to rate settlement");
                return Err(err.error).context("failed to rate the settlement")
            }
```